### PR TITLE
fix #1853 Caching the last /api/deployments

### LINKED ISSUE #1853: Caching the last /api/deployments and its etag to improve perf
Given the unchanged nature of deployment list, caching the last result of `/api/deployments` and its Etag would improve performance.   This reduced the number of file read in normal scenario to one.  We could also trace how long does it take to calculate etag (reading a file) for debugging purpose.

### PR DESCRIPTION
**Pull request by suwatch**

### CODE REVIEW

File: Kudu.Services/Deployment/DeploymentController.cs [hunk 1]
```diff
@@ -296,7 +305,18 @@ public HttpResponseMessage GetDeployResults()
             {
                 using (_tracer.Step("DeploymentService.GetDeployResults"))
                 {
-                    IEnumerable<DeployResult> results = GetResults(Request).ToList();
+                    if (!currentEtag.Equals(cachedDeployments.Etag))
```
Comment (davidebbo) [inline Kudu.Services/Deployment/DeploymentController.cs]: `EtagEquals` like above?

Comment (davidebbo) [inline Kudu.Services/Deployment/DeploymentController.cs]: Oh, I see that one takes a Request. Ignore previous comment...

File: Kudu.Services/Deployment/DeploymentController.cs [hunk 2]
```diff
@@ -296,7 +305,18 @@ public HttpResponseMessage GetDeployResults()
...
+                        cachedDeployments = new DeploymentsCacheItem
+                        {
+                            Results = GetResults(Request).ToList(),
+                            Etag = currentEtag
+                        };
+
+                        _cachedDeployments = cachedDeployments;
+                    }
+
+                    IEnumerable<DeployResult> results = cachedDeployments.Results;
```
Comment (davidebbo) [inline Kudu.Services/Deployment/DeploymentController.cs]: Somewhat pointless variable. Probably just inline in next statement.

**Comment by davidebbo** [comment]
Minor comment. :shipit:

File: Kudu.Services/Deployment/DeploymentController.cs (hunk 2 above)
Comment (suwatch) [inline Kudu.Services/Deployment/DeploymentController.cs]: Will do.

**Pull request closed without merging**